### PR TITLE
Replace inheritance from lambdas (fixes Kotlin/JS)

### PR DESCRIPTION
**Pull request by igordmn**
Fixes COMPOSE-161 1.5.0-beta03 merge. Fix FocusedBounds.kt

- Apply changes from jb-main between the base commit 6022301db806601f282c53b8cbb5a981923a1589 and the pre-merge commit 4ac3bda449d4606e5190208d074654ba041d2537:

  - d3a653ce
  
- Find new `override fun invoke` and replace lambda inheritance there

### CODE REVIEW

File: compose/ui/ui/src/commonMain/kotlin/androidx/compose/ui/focus/FocusProperties.kt [hunk 1]
```diff
@@ -182,7 +182,7 @@ fun Modifier.focusProperties(
 ): Modifier = this then FocusPropertiesElement(scope)
 
 private data class FocusPropertiesElement(
-    val scope: FocusProperties.() -> Unit
+    val scope: InvokeOnFocusProperties
```
Comment (igordmn) [inline compose/ui/ui/src/commonMain/kotlin/androidx/compose/ui/focus/FocusProperties.kt:185]: @eymar, how can we check that we need it now?

Comment (eymar) [reply compose/ui/ui/src/commonMain/kotlin/androidx/compose/ui/focus/FocusProperties.kt:185]: We can check it by running a demo app with k/js.

```kotlin
) : Modifier.Node(), ModifierLocalModifierNode, (LayoutCoordinates?) -> Unit {
```
It was not possible to implement a Function1...N interface in k/js before. I think it's still not possible. 

So we had to do such a workaround:

``` kotlin
 ) : Modifier.Node(), ModifierLocalModifierNode, InvokeOnLayoutCoordinates {
```

Comment (eymar) [reply compose/ui/ui/src/commonMain/kotlin/androidx/compose/ui/focus/FocusProperties.kt:185]: I'll quickly check it now

Comment (eymar) [reply compose/ui/ui/src/commonMain/kotlin/androidx/compose/ui/focus/FocusProperties.kt:185]: Given such a class:

```kotlin
 class Abc : () -> Unit {
     override fun invoke() {
         TODO("Not yet implemented")
     }

 }
 ```
 
 k/js shows an error: `Implementing function interface is prohibited in JavaScript`
 
 So the workaround is still needed

**Review by eymar — APPROVED** [review]

File: compose/ui/ui/src/commonMain/kotlin/androidx/compose/ui/focus/FocusProperties.kt (hunk 1 above)
Comment (igordmn) [reply compose/ui/ui/src/commonMain/kotlin/androidx/compose/ui/focus/FocusProperties.kt:185]: Thanks!

Added this workaround for the new functions as well in the last commit.

**Review by eymar — APPROVED** [review]

**Pull request merged**